database/sqlite/sqlite_functions.c

```diff
@@ -48,6 +48,31 @@ static int execute_insert(sqlite3_stmt *res)
     return rc;
 }
 
+#define MAX_OPEN_STATEMENTS (512)
+
+static void add_stmt_to_list(sqlite3_stmt *res)
+{
+    static int idx = 0;
+    static sqlite3_stmt *statements[MAX_OPEN_STATEMENTS];
+
+    if (unlikely(!res)) {
+        while (idx > 0)
+            sqlite3_finalize(statements[--idx]);
+        return;
+    }
+
+    if (unlikely(idx == MAX_OPEN_STATEMENTS))
```

Comment:
AFAICT, this should never happen in practice. Would it be okay to add an `info` when `idx == MAX_OPEN_STATEMENTS`?

Reply:
True it may not reach it and it's best effort to release memory during shutdown. 

I didn't put an info message here because we can't really make any use of it. The proper way will be to release the resources associated with the statement when that thread terminates (and take advantage of the resources during runtime)